web/src/engine/websites/decorators/GManga.ts

```diff
@@ -0,0 +1,360 @@
+    const request = new FetchRequest(new URL(`/mangas/${chapter.Identifier}`, this.URI).href);
+    const response = await FetchCSS(request, 'script[data-component-name="HomeApp"]');
+    const data: APIPages = JSON.parse(response[0].textContent);
+
+    const url = (data.globals.wla.configs.http_media_server || data.globals.wla.configs.media_server) + '/uploads/releases/';
+    let images: string[] = [];
+    if (data.readerDataAction.readerData.release.pages && data.readerDataAction.readerData.release.pages.length > 0) {
+        images = data.readerDataAction.readerData.release.pages.map(page => '/hq/' + page);
+    } else {
+        images = data.readerDataAction.readerData.release.webp_pages.map(page => '/hq_webp/' + page);
+    }
+    return images.map(image => {
+        const uri = new URL(url, request.url);
+        uri.pathname += data.readerDataAction.readerData.release.storage_key + image;
+        return new Page(this, chapter, uri);
+    });
+
+}
+
+/***********************************************
+ ******** Image Data Extraction Methods ********
+ ***********************************************/
+
+/**
+ * An extension method to get the image data for the given {@link page}.
+ * @param this - A reference to the {@link MangaScraper} instance which will be used as context for this method
+ * @param page - A reference to the {@link Page} containing the necessary information to acquire the image data
+ * @param priority - The importance level for ordering the request for the image data within the internal task pool
+ * @param signal - An abort signal that can be used to cancel the request for the image data
+ * @param detectMimeType - Force a fingerprint check of the image data to detect its mime-type (instead of relying on the Content-Type header)
+ */
+async function FetchImageAjax(this: MangaScraper, page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+    let data = await Common.FetchImageAjax.call(this, page, priority, signal);
+    if (data.type === 'text/html') {
+        const newUrl = new URL(page.Link.href.replace('/hq', '/mq'));
+        page.Link.href = newUrl.href;
+        data = await Common.FetchImageAjax.call(this, page, priority, signal);
+    }
+    return data;
+}
+
+/**
+ * A class decorator that adds the ability to get the image data for a given page by loading the source asynchronous with the `Fetch API`.
+ */
+export function ImageAjax() {
+    return function DecorateClass<T extends Common.Constructor>(ctor: T, context?: ClassDecoratorContext): T {
+        Common.ThrowOnUnsupportedDecoratorContext(context);
+
+        return class extends ctor {
+            public async FetchImage(this: MangaScraper, page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+                return FetchImageAjax.call(this, page, priority, signal);
+            }
+        };
+    };
+}
+
+export async function _haqiqa(t: string) : Promise<string>{
```

Comment:
Why is this method named `_haqiqa` ? I search on google and it means truth in arabic language, so I'm confused ^^

Reply:
Actually, since the main language is Arabic it might make sens

Reply:
I guess it was named like that in Gmanga, so it end up like that in the template in Hakuneko :D

Reply:
I think renaming this method with something like `_decrypt` is a bit less disturbing ^^

Reply:
then `decrypt` because _ prefix is meant to be used for unused stuff